### PR TITLE
Ivy eigh tridiagonal

### LINKED ISSUE #10208: eigh_tridiagonal

### PR DESCRIPTION
**Pull request by somaia02**
Close #10208

### CODE REVIEW

File: ivy/functional/ivy/experimental/linear_algebra.py [hunk 1]
```diff
@@ -20,6 +21,57 @@ def _check_valid_dimension_size(std):
...
+    beta: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    eigvals_only: bool = True,
+    select: str = 'a',
+    select_range: Optional[Union[Tuple[int, int], List[int], ivy.Array, ivy.NativeArray]] = None,
+    tol: Optional[float] = None,
+) -> Union[ivy.Array, Tuple[ivy.Array, ivy.Array]]:
+    """
+    """
```
Comment (fspyridakos) [inline ivy/functional/ivy/experimental/linear_algebra.py:40]: Could you add docstrings to the function as done with other functions, this is further explained here: https://lets-unify.ai/ivy/deep_dive/docstrings.html

File: ivy/functional/ivy/experimental/linear_algebra.py [hunk 2]
```diff
@@ -20,6 +21,57 @@ def _check_valid_dimension_size(std):
...
+            eigenvalues.less_equal(select_range[1])
+        )
+        eigenvalues = eigenvalues[condition]
+        eigenvectors = eigenvectors[:,condition]
+
+    if eigvals_only:
+        return eigenvalues
+    return eigenvalues, eigenvectors
+
+
```
Comment (fspyridakos) [inline ivy/functional/ivy/experimental/linear_algebra.py:74]: add `eigh_tridiagonal.mixed_function = True`

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 3]
```diff
@@ -134,6 +134,83 @@ def _generate_diag_args(draw):
...
+        fw=backend_fw,
+        fn_name=fn_name,
+        on_device=on_device,
+        input_dtypes=dtype,
+        alpha=alpha,
+        beta=beta,
+        eigvals_only=eigvals_only,
+        select=select,
+        select_range=select_range,
+        tol=tol,
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:210]: Might want to add some tolerance to the test like so: `rtol_=1e-2, atol_=1e-2,`

**Review by fspyridakos — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please go over the requested changes. Also there seem to be quite a few lint issues, I would reccommend you install our pre-commit hook to fix them: https://lets-unify.ai/ivy/contributing/setting_up.html#pre-commit

**Comment by somaia02** [comment]
Is it meaningful to have instance method variants of this function?

**Comment by somaia02** [comment]
Tensorflow throws an error when there is no eigenvalues in the range provided by the select_range argument. How do I deal with this?

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 4]
```diff
@@ -134,6 +135,123 @@ def _generate_diag_args(draw):
...
+        test_values=eigvals_only,
+        return_flat_np_arrays=True,
+    )
+    if results is None:
+        return
+    ret_np_flat, ret_from_np_flat = results
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:223]: The second one would come form the ground truth framework, you can abbreviate it at `gt`

**Review by fspyridakos — CHANGES_REQUESTED** [review]
Good work so far! For the select range, I would suggest you could just make it quite large such that it's guaranteed to have eigenvalues within it, let me know if that works. Also making intance methods for all the functins in the API is a design choice we explain further here: https://lets-unify.ai/ivy/design/ivy_as_a_framework/ivy_array.html#instance-methods

**Comment by fspyridakos** [comment]
LGTM! Many thanks for your contribution again!

**Pull request merged**